pkg/runtime/runtime.go

```diff
@@ -1540,12 +1560,14 @@ func (a *DaprRuntime) publishMessageGRPC(ctx context.Context, msg *pubsubSubscri
 			decoded, decodeErr := base64.StdEncoding.DecodeString(dataAsString)
 			if decodeErr != nil {
 				log.Debugf("unable to base64 decode cloudEvent field data_base64: %s", decodeErr)
+				diag.DefaultComponentMonitoring.PubsubIngressEvent(ctx, msg.pubsub, string(pubsub.Retry), msg.topic, 0)
```

Comment:
this ideally should not come under `retry` since it seems to be a data error.

Reply:
Dapr treats this as a Retry, and the monitoring simply follows the logic. This PR doesn't deal at all with how Dapr handles pub/sub errors. Will not change.